app/gui2/src/components/ComponentBrowser.vue

```diff
@@ -27,18 +28,38 @@ const props = defineProps<{
   navigator: ReturnType<typeof useNavigator>
   initialContent: string
   initialCaretPosition: ContentRange
+  sourceNode: Opt<ExprId>
 }>()
 
 const emit = defineEmits<{
   finished: [selectedExpression: string]
 }>()
 
+const initialContent = computed(() => {
+  if (props.sourceNode == null) return props.initialContent
+  const sourceNode = props.sourceNode
+  const sourceNodeName = useGraphStore().getNodeBinding(sourceNode)
+  const sourceNodeNameWithDot = sourceNodeName ? sourceNodeName + '.' : ''
+  return sourceNodeNameWithDot + props.initialContent
+})
+
+const initialCaret = computed((): ContentRange => {
+  if (props.sourceNode == null) return props.initialCaretPosition
+  const sourceNode = props.sourceNode
+  const sourceNodeName = useGraphStore().getNodeBinding(sourceNode)
```

Comment:
The store should be obtained in modules scope, like `projectStore` or `suggestionDbStore` below.